src/classes/Action.ts

```diff
@@ -39,9 +39,11 @@ enum ActivePeriod {
 class Frequency {
   public readonly Uses: number
   public readonly Duration: ActivePeriod
+  public readonly FreqText: string
   private _unlimited: boolean
 
   public constructor(frq: string) {
+    this.FreqText = frq
```

Comment:
Adding a Serialization function for Frequency felt like overkill, so I stored the original input for the constructor instead. :/

Reply:
From a consistency standpoint, this isn't the best. However, to be fully consistent we would also need an IFrequencyData struct with only one member which is a little silly.
I think this is fine until and unless we need more than one thing to specify a frequency.

Reply:
From a consistency standpoint, this isn't the best. However, to be fully consistent we would also need an IFrequencyData struct with only one member which is a little silly.
I think this is fine until and unless we need more than one thing to specify a frequency.

Reply:
Understood Re: Consistency; my feelings on this addition align with yours.